mlflow/tracking/fluent.py

```diff
@@ -359,7 +372,28 @@ def start_run(
             )
         active_run_obj = client.get_run(existing_run_id)
     else:
-        parent_run_id = _active_run_stack[-1].info.run_id if len(_active_run_stack) > 0 else None
+        if parent_run_id:
+            _validate_run_id(parent_run_id)
+            # Make sure parent_run_id matches the current run id, if there is an active run
+            if len(_active_run_stack) > 0 and parent_run_id != _active_run_stack[0].info.run_id:
+                raise Exception(
+                    (
+                        "Current run with UUID {} does not match the specified parent_run_id {}"
+                        + " To start a new nested run under the parent run with UUID {}, "
+                        + "first end the current run with mlflow.end_run()."
+                    ).format(_active_run_stack[0].info.run_id, parent_run_id)
```

Comment:
```suggestion
            if len(_active_run_stack) > 0 and parent_run_id != _active_run_stack[-1].info.run_id:
                raise Exception(
                    (
                        "Current run with UUID {} does not match the specified parent_run_id {}"
                        + " To start a new nested run under the parent run with UUID {}, "
                        + "first end the current run with mlflow.end_run()."
                    ).format(_active_run_stack[-1].info.run_id, parent_run_id)
```

Should this be consistent with the behavior of `nested=True`?